django/views/generic/list.py

```diff
@@ -181,7 +181,13 @@ def get_template_names(self):
         if hasattr(self.object_list, 'model'):
             opts = self.object_list.model._meta
             names.append("%s/%s%s.html" % (opts.app_label, opts.model_name, self.template_name_suffix))
-
+        else:
+            if self.object_list is None and not names:
+                raise ImproperlyConfigured(
+                    "Expected a queryset, but found None. Please check that "
```

Comment:
`QuerySet` instead of `queryset` for all instances of it in this exception message.

Reply:
Also rephrase to a single sentence to avoid mentioning `QuerySet` twice.